draft-ietf-quic-transport.md

```diff
@@ -2481,12 +2481,15 @@ have 1-RTT keys, so the endpoint SHOULD send CONNECTION_CLOSE frames in a
 Handshake packet.  If the endpoint does not have Handshake keys, it SHOULD send
 CONNECTION_CLOSE frames in an Initial packet.
 
-A client will always know whether the server has Handshake keys
-(see {{discard-initial}}), but it is possible that a server does not know
-whether the client has Handshake keys.  Under these circumstances, a server
-SHOULD send a CONNECTION_CLOSE frame in both Handshake and Initial packets
-to ensure that at least one of them is processable by the client.  These
-packets can be coalesced into a single UDP datagram (see {{packet-coalesce}}).
+A client will always know whether the server has Handshake keys (see
+{{discard-initial}}), but it is possible that a server does not know whether the
+client has Handshake keys.  Under these circumstances, a server SHOULD send a
+CONNECTION_CLOSE frame in both Handshake and Initial packets to ensure that at
+least one of them is processable by the client.  Similarly, a peer might be
```

Comment:
This last ambiguity is only really a problem on the client side, because the client doesn't discard Handshake keys until after the server has already discarded them(so it couldn't send a close in Handshake anyway).

Reply:
To ensure that there is a record of this.  There are situations where handshake keys are available for sending, but not receiving at both peers.

The client might have sent it's last Handshake packet, but the server can't read 1-RTT until it processes that packet (and maybe verifies a certificate).  A server might have sent its first Handshake flight, but the client hasn't received all of it, so it won't have 1-RTT keys.

Reply:
> The client might have sent it's last Handshake packet

Its such a shame: https://www.theguardian.com/uk-news/2019/dec/01/laziness-has-won-apostrophe-society-admits-its-defeat

Reply:
Yikes.  I don't make that mistake often, but you got me.

Reply:
heh, I'm just sad that the apostrophe society is dead.